docs/qsl-api.md

```diff
@@ -0,0 +1,305 @@
+# QSL API
+
+## Purpose
+
+Provide an easy to use language for users to query objects from dgraph without having to use graphQL
+
+## Format
+  `objecttype[@filtername="value"]{@field1,@field2}`
+  `objecttype1[@filtername="value"]{*}.objecttype2[@fieldname="value",@fieldname2="value2"]{*}`
+
+### Details
+  * objecttype - the kubernetes Kind of the object
+    * needs to match the case as Kubernetes would
+    * e.g. ReplicaSet will work, Replicaset/rEplicaset/REPLICASET/etc. will not
+    * will automatically capitalize first letter, so pod and Pod will both work
+  * filtername=value - filter to get objects of objecttype where the filtername = value
+    * value must be enclosed in quotes and can contain alphanumeric characters, ., -, and _
+    * , can be used as boolean equivalent to AND
+    * | can be used as the boolean equivalent to OR
+    * AND takes precedence over OR
+      * e.g. a,b,c|d,e === (a&b&c) | (d&e)
+    * other comparators (<,>,<=,>=) can be used for data types that support comparison
+      * e.g. ReplicaSet[\@numreplicas>="1"]{\*}
+  * field - the fields of the object that we want to return
+    * each field must begin with an @ followed by an alphanumeric field name, or be a string of \*
+     * the list can either only contain comma separated @-prefixed field names or \* strings, not both
+    * the "\*" will return all fields of the object in dgraph
+    * n "\*" will get all fields and relations n edges away from the node
+      * if multiple \* are present any relations after that block are ignored
+      * e.g. cluster[...]{\*\*}.namespace[...]{@name} is the same as cluster[...]{\*\*}
+  * objecttype1[...]{\*}.objecttype2[...]{\*} - the . denotes a relationship objecttype1->objecttype2
+    this will get all fields from objecttype1 and all objecttype2's related to the results of the first block
+    with all their fields
+  * objecttype, filtername/value and the fields must be nonempty
+
+## Examples
+  ```
+  pod[@name="pod1"]{@phase}
+    return the value of the phase field for all pods named pod1
+  ```
+
+  ```
+  cluster[@name="cluster1"]{*}.pod[@name="pod1",@ip="1.1.1.1"]{@phase,@image}
+    find all clusters named cluster 1
+    then find all pods named pod1 with ip 1.1.1.1 related to those clusters
+  ```
+
+  ```
+  pod[@name="pod1"]{**}
+    return the fields of pod1 and the fields for all direct relationships
+  ```
+
+## QSL Queries and Their DGraph Equivalents
+  ```
+  qsl: cluster[@name="paas-preprod-west2.cluster.k8s.local"]{@name}
```

Comment:
one thing I've been trying to be conscious of sanitizing our mock/sample data both in code and dox to make sure we're not exposing anything intuit sensitive

for instance, i lost track of whether paas is just a generally accepted acronym or something intuit specific so I tried to remove that from such strings in my code

also maybe in use of certain tools (like tiller referred to below)

what do you think?

Reply:
@kianjones4, just to confirm... discussed this with @formuzi and we don't need to consider either "paas" or the references to "tiller" as sensitive